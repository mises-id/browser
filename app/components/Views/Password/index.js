/*
 * @Author: lmk
 * @Date: 2021-07-19 12:17:48
 * @LastEditTime: 2021-07-19 22:11:07
 * @LastEditors: lmk
 * @Description: Restore misesid
 */
import { strings } from 'app/locales/i18n';
import React, { useReducer, useEffect, useRef, useState } from 'react';
import {Platform, StyleSheet, Text, View,Image ,TouchableOpacity, TextInput, Button} from 'react-native';
import radio from 'app/images/radio.png'
import radioChecked from 'app/images/radio_selected.png'
const Password = (props)=>{
  const [checked, setchecked] = useState(false)
  return <View style={styles.pageBox}>
    <View style={styles.titleBox}><Text style={styles.title}>{strings('password.title')}</Text></View>
    <View style={styles.inputTitleBox}><Text style={styles.inputTitleTxt}>{strings('password.inputTxt')}:</Text></View>
    <View style={styles.inputBox}>
      <TextInput placeholder={strings('common.placeholder')}></TextInput>
    </View>
    <View style={styles.inputTitleBox}><Text style={styles.inputTitleTxt}>{strings('password.inputTxtConfirm')}:</Text></View>
    <View style={styles.inputBox}>
      <TextInput placeholder={strings('common.placeholder')}></TextInput>
    </View>
    <View style={styles.txtBtnBox}>
      <TouchableOpacity onPress={()=>setchecked(!checked)} style={{flexDirection:'row',alignItems:'center'}}>
        <Image source={checked? radioChecked : radio} style={styles.radio}></Image>
        <Text style={styles.stay}>{strings('password.stay')}</Text>
      </TouchableOpacity>
      <Text  style={styles.txtBtnStyle}>{strings('password.forgot')}</Text>
    </View>
    <View style={styles.btnBox}>
      <View style={[styles.btnStyle,styles.success]}>
        <Text style={styles.successBtnTxt}>{strings('password.success_button')}</Text>
      </View>
    </View>
  </View>
}
const styles = StyleSheet.create({
  pageBox:{
    paddingLeft:15,
    paddingRight:15
  },
  title:{
    fontSize:16,
    color:"#333",
    fontWeight:"bold"
  },
  titleBox:{
    paddingTop:25,
    paddingBottom:17
  },
  inputBox:{
    height:50,
    backgroundColor:"#F8F8F8",
    borderRadius:5,
    paddingLeft:15,
    paddingRight:15,
    marginBottom:25
  },
  inputTitleBox:{
    marginTop:0,
    marginBottom:15
  },
  btnBox:{
    marginTop:15,
    marginLeft:25,
    marginRight:25
  },
  btnStyle:{
    borderWidth:1,
    height:50,
    borderRadius:50,
    alignItems:'center',
    justifyContent:'center',
    overflow:'hidden',
    marginTop:25
  },
  inputTitleTxt:{
    fontSize:16,
    color:'#333',
    
  },
  success:{
    borderColor:'#5D61FF'
  },
  successBtnTxt:{
    fontSize:16,
    color:"#5D61FF",
    fontWeight:'bold'
  },
  cancel:{
    borderColor:'#EEEEEE',
    backgroundColor:'#EEEEEE'
  },
  cancelBtnTxt:{
    fontSize:16,
    color:"#666",
    fontWeight:'bold'
  },
  txtBtnBox:{
    flexDirection:'row',
    justifyContent:"space-between"
  },
  txtBtnStyle:{
    color:"#5D61FF"
  },
  radio:{
    width:15,
    height:15
  },
  stay:{
    marginLeft:6,
    fontSize:13,
    color:'#666'
  }
})
Password.navigationOptions = ()=>{
  return {
    headerTitle:strings('menu.create')
  }
}
export default Password