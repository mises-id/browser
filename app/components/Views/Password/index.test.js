/*
 * @Author: lmk
 * @Date: 2021-07-19 22:12:23
 * @LastEditTime: 2021-07-19 22:12:43
 * @LastEditors: lmk
 * @Description: Password test
 */
import React from 'react';
import {shallow} from 'enzyme';
import Password from './';

describe('Password', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Password/>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
