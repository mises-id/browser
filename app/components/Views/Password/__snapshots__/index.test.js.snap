// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password should render correctly 1`] = `
<View
  style={
    Object {
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
>
  <View
    style={
      Object {
        "paddingBottom": 17,
        "paddingTop": 25,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Browser local password
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 16,
        }
      }
    >
      Password
      :
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderRadius": 5,
        "height": 50,
        "marginBottom": 25,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <TextInput
      placeholder="Please enter the contents"
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 16,
        }
      }
    >
      Confirm password
      :
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderRadius": 5,
        "height": 50,
        "marginBottom": 25,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <TextInput
      placeholder="Please enter the contents"
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/images/radio.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 15,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#666",
            "fontSize": 13,
            "marginLeft": 6,
          }
        }
      >
        Stay logged in
      </Text>
    </TouchableOpacity>
    <Text
      style={
        Object {
          "color": "#5D61FF",
        }
      }
    >
      Forgot password?
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 25,
        "marginRight": 25,
        "marginTop": 15,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 25,
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#5D61FF",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#5D61FF",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Confirm
      </Text>
    </View>
  </View>
</View>
`;
